sdk/webpubsub/azure-messaging-webpubsubclient/azure/messaging/webpubsubclient/aio/_client.py

```diff
@@ -0,0 +1,1179 @@
+        :keyword bool ack: If False, the message won't contains ackId and no AckMessage
+         will be returned from the service. Default is True.
+        :keyword bool no_echo: Whether the message needs to echo to sender. Default is False.
+        """
+
+    async def send_to_group(
+        self,
+        group_name: str,
+        content: Union[str, memoryview, Dict[str, Any]],
+        data_type: WebPubSubDataType,
+        **kwargs: Any,
+    ) -> None:
+        """Send message to group.
+        :param group_name: The group name. Required.
+        :type group_name: str.
+        :param content: The data content. Required.
+        :type content: Union[str, memoryview, Dict[str, Any]].
+        :param data_type: The data type. Required.
+        :type data_type: ~azure.messaging.webpubsubclient.models.WebPubSubDataType or str.
+        :keyword bool ack: If False, the message won't contains ackId and no AckMessage
+         will be returned from the service. Default is True.
+        :keyword bool no_echo: Whether the message needs to echo to sender. Default is False.
+        """
+
+        async def send_to_group_attempt():
+            ack = kwargs.pop("ack", True)
+            no_echo = kwargs.pop("no_echo", False)
+            if ack:
+                await self._send_message_with_ack_id(
+                    message_provider=lambda id: SendToGroupMessage(
+                        group=group_name,
+                        data_type=data_type,
+                        data=content,
+                        ack_id=id,
+                        no_echo=no_echo,
+                    ),
+                    **kwargs,
+                )
+            else:
+                await self._send_message(
+                    message=SendToGroupMessage(
+                        group=group_name,
+                        data_type=data_type,
+                        data=content,
+                        no_echo=no_echo,
+                    ),
+                    **kwargs,
+                )
+
+        await self._retry(send_to_group_attempt)
+
+    async def _retry(self, func: Callable[[], Awaitable[None]]):
+        retry_attempt = 0
+        while True:
+            try:
+                await func()
+                return
+            except Exception as e:  # pylint: disable=broad-except
+                retry_attempt = retry_attempt + 1
+                delay_seconds = self._message_retry_policy.next_retry_delay(retry_attempt)
+                if delay_seconds is None:
+                    raise e
+                _LOGGER.debug(
+                    "will retry %sth times after %s seconds",
+                    retry_attempt,
+                    delay_seconds,
+                )
+                await asyncio.sleep(delay_seconds)
+
+    def _call_back(self, callback_type: CallbackType, *args):
+        for func in self._handler[callback_type]:
+            asyncio.create_task(func(*args))
```

Comment:
Are we not tracking the futures that are being created here to make sure they're awaited on before shutdown?

Reply:
Confirmed with @chengyuanlai-msft who is author of JS/.NET SDK and also from service team that SDK should not await for task done when client stopped.

Reply:
I disagree with this premise - the client lifetime should include any threads/coroutines that it starts. Why would we leave coroutines behind?

Reply:
I see. I add `_tasks` to record all the created tasks by client and await them before client close.